quinn-proto/src/crypto.rs

```diff
@@ -102,7 +102,7 @@ pub fn reset_token_for(key: &SigningKey, id: &ConnectionId) -> [u8; RESET_TOKEN_
 pub enum Crypto {
```

Comment:
Actually, throughout this PR I have been feeling that this enum is not pulling its weight, and we should just use `CryptoContext` as the `Crypto` type. At this point, it seems we have more code working around the variants than using them, and I think usages in `Connection` are disambiguated by their usage sites. @Ralith what do you think?

@stammw I don't want to let your PR spiral out of control if you don't want it to, so do with this comment what you like.

Reply:
I'm 100% on board with ditching the enum outright.